src/lib/core.c

```diff
@@ -894,6 +894,60 @@ int gpiod_line_event_read_fd(int fd, struct gpiod_line_event *event)
 	return 0;
 }
 
+static struct gpiod_chip *
+gpiod_chip_iter_next_noclose(struct gpiod_chip_iter *iter);
+
+struct gpiod_chip_bulk * gpiod_chip_open_all(void)
+{
+	struct gpiod_chip_iter *chip_iter;
+	struct gpiod_chip_bulk *bulk;
+	struct gpiod_chip **chips;
+	struct gpiod_chip *chip;
+
+	chip_iter = gpiod_chip_iter_new();
+	if (!chip_iter)
+		return NULL;
+
+	bulk = zalloc(sizeof(*bulk));
+	if (!bulk)
+		return NULL;
+
+	for ((chip) = gpiod_chip_iter_next_noclose(chip_iter);	\
+	     !gpiod_chip_iter_done(chip_iter);			\
+	     (chip) = gpiod_chip_iter_next_noclose(chip_iter)) {
+		if (gpiod_chip_iter_err(chip_iter))
+			continue;
+
+		chips = realloc(bulk->chips,
```

Comment:
Hm, the glibc malloc implementation already does several optimizations for that, not sure if we gain much. I did not want to optimize prematurely. But we can do it anyway.